modules/claims_api/app/controllers/claims_api/application_controller.rb

```diff
@@ -43,7 +62,11 @@ def veteran_from_headers(with_gender: false)
         va_profile: ClaimsApi::Veteran.build_profile(header('X-VA-Birth-Date')),
         last_signed_in: Time.now.utc
       )
-      vet.loa = @current_user.loa if @current_user
+      vet.loa = if @current_user
+                  @current_user.loa
+                else
+                  vet.loa = { current: header('X-VA-LOA'), highest: header('X-VA-LOA') }
```

Comment:
we need to ensure that if we fall back to X-VA-LOA and it's nil that we fall back to 0 so that it fails due to permissions instead of 500 because it's a nil value.

Reply:
Thats actually taken care of by an order of operations. Before target veteran is set, we validate that the header is present and the level is 3. I dont love it because some one could call this method outside of the current work flow, but then all the required headers could fail.

Reply:
:+1: